src/main/kotlin/org/jitsi/jibri/api/xmpp/XmppApi.kt

```diff
@@ -0,0 +1,245 @@
+ * to the given [JibriManager].  The IQ messages are instances of [JibriIq] and allow the
+ * starting and stopping of the services Jibri provides.
+ * [XmppApi] subscribes to [JibriManager] status updates and translates those into
+ * XMPP presence (defined by [JibriStatusPacketExt]) updates to advertise the status of this Jibri.
+ * [XmppApi] takes care of translating the XMPP commands into the appropriate
+ * [JibriManager] API calls and translates the results into XMPP IQ responses.
+ */
+class XmppApi(
+    private val jibriManager: JibriManager,
+    private val xmppConfigs: List<XmppEnvironmentConfig>
+) {
+    private val logger = Logger.getLogger(this::class.qualifiedName)
+    private val executor = Executors.newSingleThreadExecutor(NameableThreadFactory("XmppApi"))
+
+    /**
+     * Start up the XMPP API by connecting and logging in to all the configured XMPP environments.  For each XMPP
+     * connection, we'll listen for incoming [JibriIq] messages and handle them appropriately.  Join the MUC on
+     * each connection and send an initial [JibriStatusPacketExt] presence.
+     */
+    fun start() {
+        JibriStatusPacketExt.registerExtensionProvider()
+        ProviderManager.addIQProvider(
+            JibriIq.ELEMENT_NAME, JibriIq.NAMESPACE, JibriIqProvider()
+        )
+
+        // Join all the MUCs we've been told to
+        for (config in xmppConfigs) {
+            for (host in config.xmppServerHosts) {
+                logger.info("Connecting to xmpp environment on $host with config $config")
+                val configBuilder = XMPPTCPConnectionConfiguration.builder()
+                    .setHost(host)
+                    .setXmppDomain(config.controlLogin.domain)
+                    .setUsernameAndPassword(config.controlLogin.username, config.controlLogin.password)
+                if (config.trustAllXmppCerts) {
+                    logger.info("The trustAllXmppCerts config is enabled for this domain, " +
+                            "all XMPP server provided certificates will be accepted")
+                    configBuilder.setCustomX509TrustManager(TrustAllX509TrustManager())
+                    configBuilder.setHostnameVerifier(TrustAllHostnameVerifier())
+                }
+                try {
+                    val mucClient = MucClient(configBuilder.build())
+                    mucClient.addIqRequestHandler(object : JibriSyncIqRequestHandler() {
+                        override fun handleJibriIqRequest(jibriIq: JibriIq): IQ {
+                            return handleJibriIq(jibriIq, config, mucClient)
+                        }
+                    })
+                    jibriManager.addStatusHandler { status ->
+                        logger.info("XMPP API got jibri status $status, publishing presence to connection ${config.name}")
+                        val jibriStatus = JibriPresenceHelper.createPresence(
```

Comment:
What do you think about renaming `jibriStatus` to `presence`? With the types being implicit I find it even more useful for the name to reflect the type.